hummingbot/strategy/perpetual_market_making/perpetual_market_making.pyx

```diff
@@ -654,8 +654,15 @@ cdef class PerpetualMarketMakingStrategy(StrategyBase):
                 take_profit_price = position.entry_price * (Decimal("1") + profit_spread) if position.amount > 0 \
                     else position.entry_price * (Decimal("1") - profit_spread)
                 price = market.c_quantize_order_price(self.trading_pair, take_profit_price)
-                old_exit_orders = [o for o in active_orders if (o.price != price and position.amount < 0 and o.client_order_id in self._exit_orders and o.is_buy)
-                                   or (o.price != price and position.amount > 0 and o.client_order_id in self._exit_orders and not o.is_buy)]
+                size = position.amount
+                old_exit_orders =[
+                    o for o in active_orders
+                    if (
+                        (o.price != price or o.size != size)
```

Comment:
```suggestion
                        (o.price != price or o.quantity != size)
```
Limit orders only has `quantity` property `filled_quantity` to check initial size of order and filled size respectively.
I'll suggest we don't bring in this complexity for several reasons.
I'll get back with more suggestion(s) on how we can solve the problem.